src/test/java/solvd/webtest/HomePageTest.java

```diff
@@ -1,40 +1,51 @@
 package solvd.webtest;
 
-import com.solvd.webtest.components.Result;
-import com.solvd.webtest.pages.CategoriesPage;
-import com.solvd.webtest.pages.HomePage;
-import com.solvd.webtest.pages.LoginPage;
-import com.solvd.webtest.pages.SearchResultPage;
+import com.solvd.webtest.components.ResultBase;
+import com.solvd.webtest.pages.base.CategoriesPageBase;
+import com.solvd.webtest.pages.base.HomePageBase;
+import com.solvd.webtest.pages.base.LoginPageBase;
+import com.solvd.webtest.pages.base.SearchResultPageBase;
+import com.zebrunner.carina.core.registrar.tag.TestTag;
 import com.zebrunner.carina.utils.R;
 import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.time.Duration;
 import java.util.ArrayList;
 import java.util.Comparator;
 import java.util.List;
+import java.util.Scanner;
 
 import static org.testng.Assert.*;
 
 public class HomePageTest extends AbstractTest {
     @Test
+    @TestTag(name = "platform", value = "desktop")
+    @TestTag(name = "platform", value = "mobile")
     public void emptyLoginTest() {
-        HomePage homePage = new HomePage(driver.get());
+        HomePageBase homePage = initPage(getDriver(), HomePageBase.class);
         homePage.open();
         assertTrue(homePage.isOpen(), "HomePage is not opened");
 
-        LoginPage loginPage = homePage.clickMyEbay();
+        LoginPageBase loginPage = homePage.clickMyEbay();
+        loginPage.waitUntil(l -> loginPage.isOpen(), Duration.ofMillis(3000));
         assertTrue(loginPage.isOpen(), "LoginPage is not opened");
 
         assertFalse(loginPage.tryLogin(""), "Missing login error message");
     }
 
     @Test
+    @TestTag(name = "platform", value = "desktop")
+    @TestTag(name = "platform", value = "mobile")
     public void validLoginTest() {
-        HomePage homePage = new HomePage(driver.get());
+        HomePageBase homePage = initPage(getDriver(), HomePageBase.class);
         homePage.open();
         assertTrue(homePage.isOpen(), "HomePage is not opened");
 
-        LoginPage loginPage = homePage.clickMyEbay();
+        LoginPageBase loginPage = homePage.clickMyEbay();
+        loginPage.waitUntil(l -> loginPage.isOpen(), Duration.ofMillis(3000));
         assertTrue(loginPage.isOpen(), "LoginPage is not opened");
 
         assertTrue(loginPage.tryLogin(R.CONFIG.get("user.login")), "Login error");
```

Comment:
After login you probably get redirected to some other page so you can just assert if this page was opened. Not sure about it tho, I am not user of ebay, just an idea :)

Reply:
This test just checks inputing username, page then asks for password or displays wrong username alert so there is no redirection to other page